phpBB/posting.php

```diff
@@ -1589,6 +1589,9 @@
 
 if ($generate_quote)
 {
+	// Remove attachment bbcode tags from the quoted message to avoid mixing with the new post attachments if any
+	$message_parser->message = preg_replace('#\[attachment=([0-9]+)\](.*?)\[\/attachment\]#uis', '\\2', $message_parser->message);
```

Comment:
This is the file name unlinked, to keep the discussion context I'd say. And it's the same way it looks when you quote attachments without adding new ones to new post.